### ISSUE TITLE
What waiting fail2ban? Ban after 1-2 minutes.

### ISSUE THREAD

**Issue by ibsteklodom** [fail2ban/fail2ban#2684]
### Environment:

_Fill out and check (`[x]`) the boxes which apply. If your Fail2Ban version is outdated, 
and you can't verify that the issue persists in the recent release, better seek support 
from the distribution you obtained Fail2Ban from_

- Fail2Ban version (including any possible distribution suffixes):**0.11.1**
- OS, including release name/version: **CentOS7**
- [ ] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:
Fail2ban checks /var/log/httpd/error_log for search basic auth errors events. 
I try to auth and see errrors in /var/log/httpd/error_log
But fail2ban to ban me after 1-2 minutes.

#### Steps to reproduce
Don't know. True..
```
[# fail2ban-client status web1c-main
Status for the jail: web1c-main
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     3
|  `- File list:        /var/log/httpd/error_log
`- Actions
   |- Currently banned: 0
   |- Total banned:     1
   `- Banned IP list:
```
#### Expected behavior
fail2ban to ban me after 1-2 seconds.
### Configuration, dump and another helpful excerpts
```
# fail2ban-client --dp
['set', 'syslogsocket', 'auto']
['set', 'loglevel', 'INFO']
['set', 'logtarget', 'SYSLOG']
['set', 'dbfile', '/var/lib/fail2ban/fail2ban.sqlite3']
['set', 'dbmaxmatches', 10]
['set', 'dbpurgeage', '1d']
['add', 'web1c-main', 'auto']
['set', 'web1c-main', 'usedns', 'warn']
['multi-set', 'web1c-main', 'addfailregex', ['.*auth_basic\\:error.*user.*not\\s+found.*XF\\s+<HOST>\\W', '.*auth_basic\\:error.*user.*authentication\\s+failure.*XF\\s+<HOST>\\W']]
['set', 'web1c-main', 'maxmatches', 3]
['set', 'web1c-main', 'ignoreself', True]
['set', 'web1c-main', 'bantime.increment', True]
['set', 'web1c-main', 'maxretry', 3]
['set', 'web1c-main', 'logencoding', 'auto']
['set', 'web1c-main', 'ignorecommand', '']
['set', 'web1c-main', 'findtime', '10m']
['set', 'web1c-main', 'bantime', '15m']
['set', 'web1c-main', 'addlogpath', '/var/log/httpd/error_log', 'head']
['set', 'web1c-main', 'addaction', 'web1c-main']
['multi-set', 'web1c-main', 'action', 'web1c-main', [['actionunban', "iptables -D fail2ban-web1c-main -p tcp --dport 80 -m string --algo bm --string 'X-Forwarded-For: <ip>' -j DROP"], ['actionban', "iptables -I fail2ban-web1c-main 1 -p tcp --dport 80 -m string --algo bm --string 'X-Forwarded-For: <ip>' -j DROP"], ['actionstart', 'iptables -N fail2ban-web1c-main\niptables -A fail2ban-web1c-main -j RETURN\niptables -I INPUT -p tcp --dport http -j fail2ban-web1c-main'], ['actioncheck', "iptables -n -L INPUT | grep -q 'fail2ban-web1c-main[ \\t]'"], ['actionstop', 'iptables -D INPUT -p tcp --dport http -j fail2ban-web1c-main\niptables -F fail2ban-web1c-main\niptables -X fail2ban-web1c-main'], ['protocol', 'tcp'], ['name', 'web1c-main'], ['chain', 'INPUT'], ['actname', 'web1c-main'], ['port', 'http']]]
['start', 'web1c-main']
```
### Any customizations done to /etc/fail2ban/ configuration
Own filter:
```ini
[Definition]
failregex = .*auth_basic\:error.*user.*not\s+found.*XF\s+<HOST>\W
            .*auth_basic\:error.*user.*authentication\s+failure.*XF\s+<HOST>\W
ignoreregex =
```
### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

### Relevant lines from monitored log files in question:
**/var/log/httpd/error_log**
```
[Wed Apr 08 16:30:36.134179 2020] [auth_basic:error] [pid 23896:tid [client 10.30.0.65:12332] AH01618: user 1 not found: /tszh  [XF 5.167.111.111]
[Wed Apr 08 16:30:37.600909 2020] [auth_basic:error] [pid 23990:tid [client 10.30.0.65:12386] AH01618: user 1 not found: /tszh  [XF 5.167.111.111]
[Wed Apr 08 16:30:39.000139 2020] [auth_basic:error] [pid 23991:tid [client 10.30.0.65:12438] AH01618: user 1 not found: /tszh  [XF 5.167.111.111]
[Wed Apr 08 16:30:40.408475 2020] [auth_basic:error] [pid 23992:tid [client 10.30.0.65:12548] AH01618: user 1 not found: /tszh  [XF 5.167.111.111]
```
**/var/log/fail2ban**
```
Apr  8 16:30:31 web1c-customer journal: fail2ban.server [24296]: INFO --------------------------------------------------
Apr  8 16:30:31 web1c-customer journal: fail2ban.server [24296]: INFO Starting Fail2ban v1.0.1.dev1
Apr  8 16:30:31 web1c-customer journal: fail2ban.server [24296]: DEBUG Creating PID file /var/run/fail2ban/fail2ban.pid
Apr  8 16:30:31 web1c-customer journal: fail2ban.observer [24296]: INFO Observer start...
Apr  8 16:30:31 web1c-customer journal: fail2ban.server [24296]: DEBUG Starting communication
Apr  8 16:30:31 web1c-customer journal: fail2ban.database [24296]: INFO Connected to fail2ban persistent database '/var/lib/fail2ban/fail2ban.sqlite3'
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: INFO Creating new jail 'web1c-main'
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: DEBUG Backend 'pyinotify' failed to initialize due to No module named pyinotify
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: DEBUG Backend 'gamin' failed to initialize due to No module named gamin
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: INFO Jail 'web1c-main' uses poller {}
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: DEBUG Setting usedns = warn for FilterPoll(Jail('web1c-main'))
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: DEBUG Created FilterPoll(Jail('web1c-main'))
Apr  8 16:30:31 web1c-customer journal: fail2ban.filterpoll [24296]: DEBUG Created FilterPoll
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: INFO Initiated 'polling' backend
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: DEBUG Setting usedns = warn for FilterPoll(Jail('web1c-main'))
Apr  8 16:30:31 web1c-customer journal: fail2ban.server [24296]: DEBUG   failregex: '.*auth_basic\\:error.*user.*not\\s+found.*XF\\s+<HOST>\\W'
Apr  8 16:30:31 web1c-customer journal: fail2ban.server [24296]: DEBUG   failregex: '.*auth_basic\\:error.*user.*authentication\\s+failure.*XF\\s+<HOST>\\W'
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: INFO Set banTime.increment = True
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: INFO   maxRetry: 3
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: INFO   encoding: UTF-8
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: INFO   findtime: 600
Apr  8 16:30:31 web1c-customer journal: fail2ban.actions [24296]: INFO   banTime: 900
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: INFO Added logfile: '/var/log/httpd/error_log' (pos = 1154890, hash = 12e5210a4692522bb3bf297833f8a82c)
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG Created <class 'fail2ban.server.action.CommandAction'>
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set actionunban = "iptables -D fail2ban-web1c-main -p tcp --dport 80 -m string --algo bm --string 'X-Forwarded-For: <ip>' -j DROP"
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set actionban = "iptables -I fail2ban-web1c-main 1 -p tcp --dport 80 -m string --algo bm --string 'X-Forwarded-For: <ip>' -j DROP"
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set actionstart = 'iptables -N fail2ban-web1c-main\niptables -A fail2ban-web1c-main -j RETURN\niptables -I INPUT -p tcp --dport http -j fail2ban-web1c-main'
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set actioncheck = "iptables -n -L INPUT | grep -q 'fail2ban-web1c-main[ \\t]'"
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set actionstop = 'iptables -D INPUT -p tcp --dport http -j fail2ban-web1c-main\niptables -F fail2ban-web1c-main\niptables -X fail2ban-web1c-main'
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set protocol = 'tcp'
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set name = 'web1c-main'
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set chain = 'INPUT'
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set actname = 'web1c-main'
Apr  8 16:30:31 web1c-customer journal: fail2ban.CommandAction [24296]: DEBUG   Set port = 'http'
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: DEBUG Starting jail 'web1c-main'
Apr  8 16:30:31 web1c-customer journal: fail2ban.filterpoll [24296]: DEBUG /var/log/httpd/error_log has been modified
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: DEBUG Seek to find time 1586344831.7 (2020-04-08 16:20:31), file size 1154890
Apr  8 16:30:31 web1c-customer journal: fail2ban.jail [24296]: INFO Jail 'web1c-main' started
Apr  8 16:30:31 web1c-customer journal: fail2ban.filter [24296]: DEBUG Position -1 from 1154890, found time None () within 0 seeks
Apr  8 16:30:31 web1c-customer journal: fail2ban.transmitter [24296]: DEBUG Status: ready
Apr  8 16:30:31 web1c-customer journal: fail2ban.utils [24296]: DEBUG 7f2837f65df0 -- returned successfully 0
Apr  8 16:30:36 web1c-customer journal: fail2ban.filterpoll [24296]: DEBUG /var/log/httpd/error_log has been modified
Apr  8 16:30:36 web1c-customer journal: fail2ban.filter [24296]: DEBUG Processing line with time:1586345436.0 and ip:5.167.111.111
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: INFO [web1c-main] Found 5.167.111.111 - 2020-04-08 16:30:36
Apr  8 16:31:36 web1c-customer journal: fail2ban.failmanager [24296]: DEBUG Total # of detected failures: 1. Current failures from 1 IPs (IP:count): 5.167.111.111:1
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: DEBUG Processing line with time:1586345437.0 and ip:5.167.111.111
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: INFO [web1c-main] Found 5.167.111.111 - 2020-04-08 16:30:37
Apr  8 16:31:36 web1c-customer journal: fail2ban.failmanager [24296]: DEBUG Total # of detected failures: 2. Current failures from 1 IPs (IP:count): 5.167.111.111:2
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: DEBUG Processing line with time:1586345439.0 and ip:5.167.111.111
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: INFO [web1c-main] Found 5.167.111.111 - 2020-04-08 16:30:39
Apr  8 16:31:36 web1c-customer journal: fail2ban.failmanager [24296]: DEBUG Total # of detected failures: 3. Current failures from 1 IPs (IP:count): 5.167.111.111:3
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: DEBUG Processing line with time:1586345440.0 and ip:5.167.111.111
Apr  8 16:31:36 web1c-customer journal: fail2ban.filter [24296]: INFO [web1c-main] Found 5.167.111.111 - 2020-04-08 16:30:40
Apr  8 16:31:36 web1c-customer journal: fail2ban.failmanager [24296]: DEBUG Total # of detected failures: 4. Current failures from 1 IPs (IP:count): 5.167.111.111:1
Apr  8 16:31:36 web1c-customer journal: fail2ban.filterpoll [24296]: DEBUG /var/log/httpd/error_log has been modified
Apr  8 16:31:36 web1c-customer journal: fail2ban.actions [24296]: NOTICE [web1c-main] Ban 5.167.111.111
Apr  8 16:31:36 web1c-customer journal: fail2ban.utils [24296]: DEBUG 7f2837f11670 -- returned successfully 0
Apr  8 16:31:36 web1c-customer journal: fail2ban.actions [24296]: DEBUG Banned 1 / 1, 1 ticket(s) in 'web1c-main'
Apr  8 16:31:37 web1c-customer journal: fail2ban.observer [24296]: DEBUG [web1c-main] Observer: failure found 5.167.111.111
Apr  8 16:31:37 web1c-customer journal: fail2ban.observer [24296]: DEBUG [web1c-main] Observer: failure found 5.167.111.111
Apr  8 16:31:37 web1c-customer journal: fail2ban.observer [24296]: DEBUG [web1c-main] Observer: failure found 5.167.111.111
Apr  8 16:31:37 web1c-customer journal: fail2ban.observer [24296]: DEBUG [web1c-main] Observer: failure found 5.167.111.111
Apr  8 16:31:37 web1c-customer journal: fail2ban.observer [24296]: DEBUG [web1c-main] Observer: ban found 5.167.111.111, 900
```

**Comment by sebres** [comment]
I see indeed a pause of exactly 1 minute between failure occurrence and `Found ...` message (as fail2ban considering that):
```
LOG | 16:30:36.134179 [auth_basic:error] [pid 23896:tid [client 10.30.0.65:12332] AH01618: user 1 not found: /tszh  [XF 5.167.111.111]
F2B | 16:30:36 web1c-customer journal: fail2ban.filterpoll [24296]: DEBUG /var/log/httpd/error_log has been modified
F2B | 16:30:36 web1c-customer journal: fail2ban.filter [24296]: DEBUG Processing line with time:1586345436.0 and ip:5.167.111.111
...
F2B | 16:31:36 web1c-customer journal: fail2ban.filter [24296]: INFO [web1c-main] Found 5.167.111.111 - 2020-04-08 16:30:36
```
No idea excepting perhaps a bottleneck fixed in b43dc147b5177019a1dcb51de9833139e904cc21 and some previous fixes, so maybe you could try to upgrade your 0.11th version to current one...
But I'm unsure it would really help in your case, because after a closer look, I see only one thing between `Processing line with time` and `Found` messages (where you have this pause):
https://github.com/fail2ban/fail2ban/blob/00c5d33e458bb8de489a3a72153bc8326d09ded2/fail2ban/server/filter.py#L626-L627

This is the check whether the found IP belonging to ignored IPs, which considering `ignoreself`, `ignoreip` and `ignorecommand`.
In your case only `ignoreself` seems to be active (this is also enabled per default), so fail2ban have to obtain own IP address to compare it with found IP.
It looks like you have some misconfiguration (DNS, hostnames, etc) which causes this timeout up to 1 minute.
Normally it should not bother a couple of further findings hereafter (because fail2ban server caches it internally for 5 minutes), so this could happen again in soonest 5 minutes.
To be sure you are affected by that, try to execute:
```
$ fail2ban-python -c 'import time; from fail2ban.server.ipdns import DNSUtils; s=time.time(); names=DNSUtils.getSelfNames(); print("%s seconds\nnames: %s" % (time.time() - s, names)); s=time.time(); ips=DNSUtils.getSelfIPs(); print("%s seconds\nips: %s" % (time.time() - s, ips))'
# resulting to:
0.000179052352905 seconds
names: set(['myhost.fqdn.tld', 'localhost'])
0.273033857346 seconds
ips: set(['192.0.2.1', '127.0.0.1', '::1'])
```
How long are the times in your case and do you see some "strange" name or IP (which theoretically may be a reason for too long DNS resolving)?
To fix that:
- either try to correct the configuration of your host (or try to speedup DNS lookup for you hosts, cache it, etc pp);
- or disable `ignoreself` (set it to false in your jail.local) and specify fixed IPs to be ignored via `ignoreip` (as comma-separated list).

**Comment by ibsteklodom** [comment]
> I see indeed a pause of exactly 1 minute between failure occurrence and `Found ...` message (as fail2ban considering that):
> 
> ```
> LOG | 16:30:36.134179 [auth_basic:error] [pid 23896:tid [client 10.30.0.65:12332] AH01618: user 1 not found: /tszh  [XF 5.167.111.111]
> F2B | 16:30:36 web1c-customer journal: fail2ban.filterpoll [24296]: DEBUG /var/log/httpd/error_log has been modified
> F2B | 16:30:36 web1c-customer journal: fail2ban.filter [24296]: DEBUG Processing line with time:1586345436.0 and ip:5.167.111.111
> ...
> F2B | 16:31:36 web1c-customer journal: fail2ban.filter [24296]: INFO [web1c-main] Found 5.167.111.111 - 2020-04-08 16:30:36
> ```
> 
> No idea excepting perhaps a bottleneck fixed in [b43dc14](https://github.com/fail2ban/fail2ban/commit/b43dc147b5177019a1dcb51de9833139e904cc21) and some previous fixes, so maybe you could try to upgrade your 0.11th version to current one...
> But I'm unsure it would really help in your case, because after a closer look, I see only one thing between `Processing line with time` and `Found` messages (where you have this pause):
> https://github.com/fail2ban/fail2ban/blob/00c5d33e458bb8de489a3a72153bc8326d09ded2/fail2ban/server/filter.py#L626-L627
> 
> This is the check whether the found IP belonging to ignored IPs, which considering `ignoreself`, `ignoreip` and `ignorecommand`.
> In your case only `ignoreself` seems to be active (this is also enabled per default), so fail2ban have to obtain own IP address to compare it with found IP.
> It looks like you have some misconfiguration (DNS, hostnames, etc) which causes this timeout up to 1 minute.
> Normally it should not bother a couple of further findings hereafter (because fail2ban server caches it internally for 5 minutes), so this could happen again in soonest 5 minutes.
> To be sure you are affected by that, try to execute:
> 
> ```
> $ fail2ban-python -c 'import time; from fail2ban.server.ipdns import DNSUtils; s=time.time(); names=DNSUtils.getSelfNames(); print("%s seconds\nnames: %s" % (time.time() - s, names)); s=time.time(); ips=DNSUtils.getSelfIPs(); print("%s seconds\nips: %s" % (time.time() - s, ips))'
> # resulting to:
> 0.000179052352905 seconds
> names: set(['myhost.fqdn.tld', 'localhost'])
> 0.273033857346 seconds
> ips: set(['192.0.2.1', '127.0.0.1', '::1'])
> ```
> 
> How long are the times in your case and do you see some "strange" name or IP (which theoretically may be a reason for too long DNS resolving)?
> To fix that:
> 
> * either try to correct the configuration of your host (or try to speedup DNS lookup for you hosts, cache it, etc pp);
> * or disable `ignoreself` (set it to false in your jail.local) and specify fixed IPs to be ignored via `ignoreip` (as comma-/space-separated list).


I disabled ignoreself. That helped. Thank you very much. It is resolved.

**Comment by sebres** [comment]
OK, then closed.
